### PR TITLE
Install Generator

### PR DESCRIPTION
**Pull request by williamkennedy**
Hi All,

I love the gem. I noticed that there was no generator, and since I never built one before, I thought I would have a go at it. 

This pull request is my proposal. Any feedback is appreciated.

These instructions were based on my experience using the gem last week, where I [noted everything down](https://williamkennedy.ninja/rails/2021/07/25/how-to-setup-storybook-with-rails-view-components-in-under-10-minutes/). I know this gem is developing fast, so I will completely understand if this misses the mark. 

Thanks for all your work,
William

P.S.

To run the generator

```
rails g view_component_storybook:install
```

### CODE REVIEW

File: lib/rails/generators/view_component_storybook/install_generator.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  module Generators
+    class InstallGenerator < Rails::Generators::Base
+
+      source_root File.expand_path("../templates", __FILE__)
+
+      desc "Installs storybook"
+
+      def install
+        insert_into_file '.gitignore', '*/*.stories.json'
+        run_yarn
```
Comment (hjhart) [inline lib/rails/generators/view_component_storybook/install_generator.rb:13]:
```suggestion
        add_yarn_dependencies
```

**Review by hjhart — COMMENTED** [review]
Nice! I've never actually seen how these are built before. Looks good. Just a suggestion on one of the method names to make it more descriptive.

**Comment by hjhart** [comment]
Awesome, William! Different, but relevant issue in case you are interested: https://github.com/jonspalmer/view_component_storybook/issues/42

**Comment by williamkennedy** [comment]
> Nice! I've never actually seen how these are built before. Looks good. Just a suggestion on one of the method names to make it more descriptive.

Done. 

I've never done of these before. Thinking about it now, there are two things I would love to be able to do

- dynamically set the port number in the preview file as some developers use port 5000 by default
- dynamically set the main.js to point at spec or test depending on which suite a developer uses.

> Awesome, William! Different, but relevant issue in case you are interested: #42

I can certainly help out on that issue, but I think it would be better to separate from this pull request.

**Comment by jonspalmer** [comment]
This is really great stuff thanks for digging in on this. However, I wonder if the Storybook specific config is better/already solved with the yarn initialization logic in Storybook. See: https://github.com/storybookjs/storybook/tree/next/app/server

roughly its: `npx sb init -t server`

I don't think that gets the .gitignore stuff and might missing other things but It's likely a better starting point for things like main.js etc.

we definitely do need a rails generator for #42.

**Comment by williamkennedy** [comment]
Thanks for the suggestion. 

This reduced the lines of code needed plus it's more stable with the Storybook API.

**Comment by williamkennedy** [comment]
@jonspalmer Are you planning to merge this in?

If not, let me know and I can help out with #42 

If so, let me know what you need to get it over the line

**Comment by jonspalmer** [comment]
@williamkennedy I'm pretty slammed with other things and I'm unfamiliar with generators myself so just need a bit of head space to think around this. 

One thing I'm wondering is is this doing enough to be more useful than adding to the instructions to use the sbx. (All this is something you do only once so its a bit different than a component generator that you do multiple times and involves a bunch more files). Also we're adding the controls dependency but not adding controls to main.js. I suspect thats difficult to automate.

**Comment by jonspalmer** [comment]
BTW saw this in your blog post

>Another Gotcha, I ran into. After you update the HTML for the component, you have to rerun the rake task.

I don't believe that to be necessary. If the stories change you need to generate new json. The HTML gets fetched each time Storybook renders the story. It will hot reload if you just touch the json files but reloading storybook or clicking between stories and back again will load new HTML.

**Comment by williamkennedy** [comment]
@jonspalmer 

Thanks for coming back to me. I see your point about the installer. My main motivation here was to improve developer onboading. I just love it when there is an install command. I didn't actually know about the npx command having never setup storybook before. 

However, I'm constantly starting new projects and I would love to be able to add this to my rails template. Hence, the installer would be used by me more than once 😂. 


With regards to the blog post, these are notes that I took as I was setting up. I'll go back and check the reloading. Thanks for pointing this out.

**Comment by jonspalmer** [comment]
I've added this feature to the storybook installer that adds the controls addon via the installer: https://github.com/storybookjs/storybook/pull/15790

Also the storybook server installer add the yarn dependencies so the second step of this installer isn't needed.

Honestly when I look at what is left I have a hard time thinking that a generator that wraps just two steps isn't really worth the indirection. Especially given that the git ignore step hardcodes a path that consumers might later tweak.

What do you think?

**Comment by williamkennedy** [comment]
Hey @jonspalmer 

Hope you're well.

Sorry I didn't back to this sooner. Just became busy.

Having looked at the recent change made to the documentation, this is not needed as the steps are pretty clear. 

We can use lessons learned about generators for the other issue discussed.

Let's close this PR if that's OK with you.

Thanks for the awesome project,
William

**Pull request closed without merging**